cmd/create.go

```diff
@@ -29,7 +29,10 @@ var componentCreateCmd = &cobra.Command{
 	Short: "Create a new component",
 	Long: `Create a new component to deploy on OpenShift.
 
-If component name is not provided, component type value will be used for the name.
+If component name is not provided, a random name $dir-$component_type-$random-4-char-str, where $dir  will be generated as per the following:
+Create using git url: repo name is $dir
+Create using local path: Directory name of the source in the local path
+Create using binary path: The name of the binary without its extension
```

Comment:
All of this input makes `odo create --help` command *way* too long. We don't need to explain everything.. See other comment, we should only have 1 sentence explaining that the name will be autogenerated.

Reply:
Ack